frontend/src/metabase/parameters/components/ValuesSourceModal/ValuesSourceTypeModal.tsx

```diff
@@ -98,7 +95,7 @@ const ValuesSourceTypeModal = ({
       {sourceType === null ? (
         <FieldSourceModal
           // if sourceType === null the parameter must have fields
-          parameter={parameter as FieldFilterUiParameter}
```

Comment:
There's no way to prove parameter has fields with the type checker so I've removed it